### PR TITLE
chore(vscode): add useVSCodeProxy option to control whether to use Vscode Proxy

### LINKED ISSUE #3360: The proxy server in vscode settings, blocks access to tabby on localhost
If proxy is enabled in vs code (http.proxy option), and tabby is installed locally, then it is impossible to connect to it.

vscode config:
```
    "http.proxyStrictSSL": false,
    "http.proxyAuthorization": null,
    "http.proxySupport": "on",
    "http.proxy": "http://10.222.3.4:2080",
```

tabby started:
```
Version 0.19.0
🚀 Listening at http://0.0.0.0:8080
```

tabby logs:
```
024-11-02 00:22:01.498 [info] [Info  - 12:22:01 AM] [Proxy] Using proxy http://10.222.3.4:2080.
2024-11-02 00:22:01.519 [info] [Debug - 12:22:01 AM] [TabbyApiClient] Health check request: GET http://127.0.0.1:8080/v1/health. [de789c23-695c-45d1-8851-600cb4450146]
2024-11-02 00:22:01.520 [info] [Debug - 12:22:01 AM] [TabbyLSP] Internal components initialized.
2024-11-02 00:22:01.521 [info] [Debug - 12:22:01 AM] [TabbyLSP] Initializing feature components...
2024-11-02 00:22:01.521 [info] [Info  - 12:22:01 AM] [RecentlyChangedCodeSearch] Created code search engine for recently changed files.
2024-11-02 00:22:01.522 [info] [Debug - 12:22:01 AM] [TabbyLSP] Feature components initialized.
2024-11-02 00:22:01.524 [info] [Info  - 12:22:01 AM] [TabbyLSP] Initialize done.
2024-11-02 00:22:01.528 [info] [Tabby] Tabby extension activated.
2024-11-02 00:22:01.528 [info] [Error - 12:22:01 AM] [TabbyApiClient] Health check request failed. [de789c23-695c-45d1-8851-600cb4450146] {
  "error": {}
}
```

### PR DESCRIPTION
**Pull request by antimonyGu**
# Test
- Setting 
```
    "tabby.settings.advanced": {
        "useVSCodeProxy": false
    },
```
, `proxy.url` will be ""(empty string).

- 
```
    "tabby.settings.advanced": {
        "useVSCodeProxy": true
    },
```
, `proxy.url` will be "http://localhost:9000".

close #3360

### CODE REVIEW

File: clients/vscode/src/Config.ts [hunk 1]
```diff
@@ -168,6 +168,8 @@ export class Config extends EventEmitter {
     const https = workspace.getConfiguration("https");
     const httpsProxy = https.get("proxy", "");
     const httpProxy = this.httpConfig.get("proxy", "");
+    // noProxy bypass
+    if (this.noProxy.includes(httpsProxy) || this.noProxy.includes(httpProxy)) return "";
```
Comment (wsxiaoys) [inline clients/vscode/src/Config.ts:172]: move to function `buildClientProvidedConfig`

File: clients/vscode/src/Config.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { getLogger } from "./logger";
 interface AdvancedSettings {
   "inlineCompletion.triggerMode"?: "automatic" | "manual";
   "chatEdit.history"?: number;
+  disableProxy?: boolean;
```
Comment (wsxiaoys) [inline clients/vscode/src/Config.ts:9]:
```suggestion
  useVSCodeProxy?: boolean;
```

File: clients/tabby-agent/src/http/proxy.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export function createProxyForUrl(url: string, configs: ProxyConfig[]): Dispatch
       logger.info("Using proxy from environment variables.");
       return new EnvHttpProxyAgent();
     } else if ("url" in config && config["url"]) {
+      // "item1, item2  item3".split(/,|\s+/). Results in: ["item1", "item2", "item3"]
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/http/proxy.ts:29]:
```suggestion
```

File: clients/vscode/src/Config.ts [hunk 4]
```diff
@@ -73,6 +74,11 @@ export class Config extends EventEmitter {
     }
   }
 
+  get useVSCodeProxy(): boolean {
+    const advancedSettings = this.workspace.get("settings.advanced", {}) as AdvancedSettings;
+    return advancedSettings["useVSCodeProxy"] || true;
```
Comment (icycodes) [inline clients/vscode/src/Config.ts:79]:
```suggestion
    return advancedSettings["useVSCodeProxy"] ?? true;
```
`anything || true` will always return `true`.

**Review by icycodes — CHANGES_REQUESTED** [review]
Please update the PR description.

**Review by icycodes — APPROVED** [review]

**Pull request merged**